pkg/genericapiserver/config.go

```diff
@@ -343,48 +352,77 @@ func (c Config) New() (*GenericAPIServer, error) {
 	}
 
 	s.installAPI(&c)
-	s.Handler, s.InsecureHandler = s.buildHandlerChains(&c, http.Handler(s.Mux.BaseMux().(*http.ServeMux)))
+
+	// Wire the handler chain:
+	// - requests from InsecureHandler are marked with skipAuth in the context
```

Comment:
At first blush, I don't think I'm a fan of this approach.  I don't like trying to apply authn/authz on the basis of a flag in a mutable context.  Handlers shouldn't be choosing whether to enforce or not.  The caller wiring them decides.

@liggitt This isn't something we already do, right?  This would be new.